invictus2/obc/include/radio_commands.h

```diff
@@ -0,0 +1,350 @@
+#ifndef RADIO_COMMANDS_H_
+#define RADIO_COMMANDS_H_
+
+#include "zephyr/toolchain.h"
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// REVIEW: make this KConfig param?
+#define SUPPORTED_PACKET_VERSION 1
+
+// -----------------------------------------------------------------------------
+// Packet header (4 bytes)
+// -----------------------------------------------------------------------------
+
+// Always 1 byte fields, packed for on-the-wire transmission.
+struct cmd_header_s {
+    uint8_t packet_version;
+    uint8_t sender_id;
+    uint8_t target_id;
+    uint8_t command_id; // use values from enum radio_command_e
+} __attribute__((packed));
```

Comment:
This might be a good way of saving space when transmitting data, but we need to be careful:

>  https://www.gnu.org/software/c-intro-and-ref/manual/html_node/Packed-Structures.html
Use of __attribute__((packed)) often results in fields that don’t have the normal alignment for their types. Taking the address of such a field can result in an invalid pointer because of its improper alignment. Dereferencing such a pointer can cause a SIGSEGV signal on a machine that doesn’t, in general, allow unaligned pointers.